### PR TITLE
Mark k8s-workload-registrar as deprecated

### PR DESCRIPTION
**Pull request by azdagron**
Fixes: #3501

### CODE REVIEW

File: support/k8s/k8s-workload-registrar/config_reconcile.go [hunk 1]
```diff
@@ -69,6 +69,8 @@ func (c *ReconcileMode) Run(ctx context.Context) error {
 	}))
 	setupLog := ctrl.Log.WithName("setup")
 
+	setupLog.Info("The k8s-workload-registrar is deprecated and no longer maintained. Please migrate to the SPIRE Controller Manager (https://github.com/spiffe/spire-controller-manager).")
```
Comment (amartinezfayo) [inline support/k8s/k8s-workload-registrar/config_reconcile.go:72]: should be at Warning level instead of Info?

Comment (azdagron) [reply support/k8s/k8s-workload-registrar/config_reconcile.go:72]: `logr` loggers only have info and error levels. I considered logging at error level but in the end determined this was more info than error :)

Comment (amartinezfayo) [reply support/k8s/k8s-workload-registrar/config_reconcile.go:72]: Fair enough.

**Review by amartinezfayo — APPROVED** [review]
Thanks @azdagron!

**Pull request merged**